lib/Rfc1035StubResolver.php

```diff
@@ -494,4 +512,19 @@ private function assertAcceptableResponse(Message $response)
             throw new DnsException(\sprintf("Server returned error code: %d", $response->getResponseCode()));
         }
     }
+
+    private function getNameserver(int $attempt = 0): string
+    {
+        $nameservers = $this->config->getNameservers();
+        $nameserversCount = \count($nameservers);
+
+        if ($this->config->shouldRotate()) {
+            $nameserver = $nameservers[$this->nextNameserver];
+            $this->nextNameserver = ++$this->nextNameserver % $nameserversCount;
```

Comment:
This should take the attempt into account, otherwise multiple attempts for one query might it the same server, which might be down, resulting in a failure even of other servers are up.

Reply:
@kelunik what do you propose? If `isRotationEnabled()` then simply select next nameserver, if not select based on attempt. I've changed in resolver that it keeps selected nameserver when retrying the same nameserver due to down or etc.

Reply:
I guess we could make `selectNameservers()` return an array of nameservers and then cycle through them as before. If rotation is enabled, a randomized or rotated array can be returned.

Reply:
@kelunik done, calling `selectNameservers()` return unchanged array or rotated if `isRotationEnabled()`